CASCEdit/Handlers/DownloadHandler.cs

```diff
@@ -50,18 +50,32 @@ public DownloadHandler(BLTEStream blte)
 					NumTags = br.ReadUInt16BE(),
 				};
 
+				if (Header.Version > 2) {
+					throw new NotImplementedException("Download file versions newer than 2 are not supported.");
+				}
+
+				if (Header.Version > 1) {
+					Header.NumFlags = br.ReadByte();
+				}
+
 				// entries
 				for (int i = 0; i < Header.NumEntries; i++)
 				{
 					var entry = new DownloadEntry()
 					{
-						Unknown = Header.Version > 1 ? br.ReadByte() : (byte)0, // new V2 field
 						Hash = new MD5Hash(br),
 						FileSize = br.ReadUInt40BE(),
-						Stage = br.ReadByte(),
-						UnknownData = br.ReadBytes(4)
+						Stage = br.ReadByte()
 					};
 
+					if (Header.Unknown != 0) {
+						entry.Unknown = br.ReadUInt32BE();
+					}
+
+					if (Header.Version > 1) {
```

Comment:
Personal style: This is one of the rare cases where I prefer `≥ n` over `> n-1`.